src/Plugin/Visualisation/Style/VisualisationStyleBase.php

```diff
@@ -334,20 +334,19 @@ protected function getSourceFieldOptions() {
    *
    * @param string $field_id
    *   The field ID.
+   * @param array $records
+   *   The set of records that we should get the values from.
    *
    * @return array
    *   The source field values.
    */
-  protected function getSourceFieldValues($field_id) {
```

Comment:
@WilsonWu2100 this was the culprit, not that array_merge. All records were getting added to `$values`, now it gets passed the correct set of records to act on

Reply:
@jez500 The fix looks pretty good and it seems the correct way to solve the problem, I am a bit scary of touching the base code, which might break something else. But anyway, thanks for your help.